app/models/user.rb

```diff
@@ -25,12 +26,32 @@ def remember
     update_attributes remember_digest: User.digest(remember_token)
   end
 
-  def authenticated? remember_token
-    return false if remember_digest.nil?
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
+    return false if digest.nil?
+    BCrypt::Password.new(digest).is_password? token
   end
 
   def forget
     update_attributes remember_digest: nil
   end
+
+  def activate
+    update_attributes activated: true, activated_at: Time.zone.now
+  end
+
+  def send_activation_email
+    UserMailer.account_activation(self).deliver_now
+  end
+
+  private
+
+  def downcase_email
+    email.downcase!
```

Comment:
can you tell me that what is different of `.downcase!` and `.downcase`?

Reply:
downcase! work on itself. downcase return a new string.

Reply:
@VuThiTranVan Yeah, I've known mail.downcase! return nil.

Reply:
ukm, so, can you fix it. you shouldn't use `downcase!`
`email.downcase!` => `self.email = email.downcase`

Reply:
@VuThiTranVan I read some comment in others' pull request that suggest use email.downcase!. :(